packages/shared/src/launchDarklylient.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  LDClient,
+  LDContext,
+  LDUser,
+  initialize as initializeLDClient,
+} from "launchdarkly-node-client-sdk";
+import { AuthInfo } from "./authentication/types";
+import { getReplayPath } from "./getReplayPath";
+import { AuthenticatedTaskQueue } from "./session/AuthenticatedTaskQueue";
+
+class LaunchDarklyClient extends AuthenticatedTaskQueue {
```

Comment:
I know it's pretty subjective so feel free to ignore this. I feel like this structure with subclassing doesn't give us a lot but it makes the control flow a little bit harder to reason about. It's also, in general, less flexible. To use `AuthenticatedTaskQueue` one has to implement the consumer as a class. I probably would refactor this to a class that anything could instantiate and introduce `queue` as a private property to all of the existing consumers. I tinkered with this locally and it felt better to me but YMMV.

Reply:
That's fair. I think I went with a class because 2 of our 3 authenticated _things_ were already classes (following the `logger` pattern, for consistency) but lemme play with this for a few and see.

Reply:
I cam up with this when I played around with it: https://github.com/replayio/replay-cli/commit/a57b91d667c0280dce6adf72902884082af56ee5 . I didn't run tests or anything but types were OK with it IIRC

Reply:
Thanks for sharing the patch, but I've already been working on a rewrite of this locally so I'm going to just follow through with it. :)

Reply:
Update: 9c59dbc

Specifically:
https://github.com/replayio/replay-cli/blob/9c59dbce9584b3ce148ce9e44a88047c6453c5ee/packages/shared/src/session/createTaskQueue.ts

Even just looking at the tests, I am convinced that this is a nicer coding mechanism:
https://github.com/replayio/replay-cli/blob/9c59dbce9584b3ce148ce9e44a88047c6453c5ee/packages/shared/src/session/createTaskQueue.test.ts